### ISSUE TITLE
city labels should be biased towards a direction (east/west?) based on bodies of water or proximity to other cities

### ISSUE THREAD

**Issue by bdon** [protomaps/basemaps#6]
This needs to be done in a general way i.e. without regards to localization, as the names may be different lengths in different languages. Think about a slimmed down version of the https://github.com/migurski/Dymo approach.

**Comment by bdon** [comment]
This doesn't actually work once you factor in map rotation, unless the rotation is available through the Style Spec.

**Comment by nvkelso** [comment]
I'm make an unsupported claim that more than 90% of users don't actually rotate the map...

**Comment by bdon** [comment]
Some more detail about what this would look like in practice:

When post-processing  the places layer we have access to other data in memory, we can use that data to modify the default value of [`text-variable-anchor`](https://maplibre.org/maplibre-style-spec/layers/#layout-symbol-text-variable-anchor) in the style

Cities on the west coast of California can have `['left','right']`, Cities on the east coast of Florida have `['right',left']`, etc

Can help the labels look a bit more cleaned up at low zooms

**Comment by nvkelso** [comment]
Another option, if this is just for low zooms, is to add the city label text orientation to Natural Earth?

**Comment by bdon** [comment]
@nvkelso yeah, that could work too! Otherwise we would need some weighting algorithm - probably based on position relative to water bodies

**Comment by bdon** [comment]
I'm going to close this for now simply because there are many higher priority things to work on. We can revisit this in the future.
